src/main/java/core/basesyntax/library/Injector.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.library;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor declaredConstructor = clazz.getDeclaredConstructor();
+        Object instance = declaredConstructor.newInstance();
+        Field[] declaredField = clazz.getDeclaredFields();
+        for (Field field : declaredField) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
```

Comment:
move it `field.setAccessible(true);` after check if (field.getType().equals(BetDao.class))